src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIImageClient.cs

```diff
@@ -0,0 +1,250 @@
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents an <see cref="IImageClient"/> for an OpenAI <see cref="OpenAIClient"/> or <see cref="ImageClient"/>.</summary>
+internal sealed class OpenAIImageClient : IImageClient
+{
+    /// <summary>Metadata about the client.</summary>
+    private readonly ImageClientMetadata _metadata;
+
+    /// <summary>The underlying <see cref="ImageClient" />.</summary>
+    private readonly ImageClient _imageClient;
+
+    /// <summary>Initializes a new instance of the <see cref="OpenAIImageClient"/> class for the specified <see cref="ImageClient"/>.</summary>
+    /// <param name="imageClient">The underlying client.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="imageClient"/> is <see langword="null"/>.</exception>
+    public OpenAIImageClient(ImageClient imageClient)
+    {
+        _ = Throw.IfNull(imageClient);
+
+        _imageClient = imageClient;
+
+        // https://github.com/openai/openai-dotnet/issues/215
+        // The endpoint and model aren't currently exposed, so use reflection to get at them, temporarily. Once packages
+        // implement the abstractions directly rather than providing adapters on top of the public APIs,
+        // the package can provide such implementations separate from what's exposed in the public API.
+        Uri providerUrl = typeof(ImageClient).GetField("_endpoint", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+            ?.GetValue(imageClient) as Uri ?? OpenAIClientExtensions.DefaultOpenAIEndpoint;
+
+        _metadata = new("openai", providerUrl, _imageClient.Model);
+    }
+
+    /// <inheritdoc />
+    public async Task<ImageResponse> GenerateImagesAsync(ImageRequest request, ImageOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(request);
+
+        string? prompt = request.Prompt;
+        _ = Throw.IfNull(prompt);
+
+        // If the request has original images, treat this as an edit operation
+        if (request.OriginalImages != null)
+        {
+            ImageEditOptions editOptions = ToOpenAIImageEditOptions(options);
+            string? fileName = null;
+            Stream? imageStream = null;
+
+            foreach (AIContent originalImage in request.OriginalImages)
+            {
+                if (imageStream is not null)
+                {
+                    // We only support a single image for editing.
+                    Throw.ArgumentException(
+                        "Only a single image can be provided for editing.",
+                        nameof(request));
+                }
+
+                if (originalImage is DataContent dataContent)
+                {
+                    imageStream = MemoryMarshal.TryGetArray(dataContent.Data, out var array) ?
+                        new MemoryStream(array.Array!, array.Offset, array.Count) :
+                        new MemoryStream(dataContent.Data.ToArray());
```

Comment:
We unfortunately do not ship a Stream that wraps a ReadOnlyMemory, though we use one internally in the .NET libraries in a few places (like HttpClient). We could source copy such a thing here, and then this wouldn't need to allocate another array. That said, `ROM<byte>` around anything other than `byte[]` is rare enough that it's probably not warranted right now.

Reply:
Yeah, I just copied this from existing cases.
I think it is more important to solve https://github.com/dotnet/extensions/issues/6683 which you previously mentioned in https://github.com/dotnet/extensions/issues/5719.  Right now our content model is all in-memory.